mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -1000,11 +1011,52 @@ convertOmpWsLoop(Operation &opInst, llvm::IRBuilderBase &builder,
   return success();
 }
 
+/// A RAII class that on construction replaces the region arguments of the
+/// parallel op (which correspond to private variables) with the actual private
+/// variables they correspond to. This prepares the parallel op so that it
+/// matches what is expected by the OMPIRBuilder. Instead of editing the
+/// original op in-place, this function does the required changes to a cloned
+/// version which should then be erased by the caller.
```

Comment:
If I understood correctly, this class is intended to make in-place changes to the operation, and not a clone. I guess this part of the comment was copied from the previous approach.

Reply:
Indeed, copy-pasta 🍝. Thanks for the diligent review.